research/deeplab/utils/train_utils.py

```diff
@@ -99,7 +99,7 @@ def get_model_init_fn(train_logdir,
   tf.logging.info('Initializing model from path: %s', tf_initial_checkpoint)
 
   # Variables that will not be restored.
-  exclude_list = ['global_step']
+  exclude_list = ['global_step', 'logits']
```

Comment:
There are cases where you do want to restore `logits' weights (e.g., further fine-tuning them on validation set).
My proposal will be: Could you please

1. Modify get_extra_layer_scopes in model.py as follows.

def get_extra_layer_scopes(last_layers_contain_logits_only=False):
  """Gets the scopes for extra layers.

  Args:
    last_layers_contain_logits_only: Boolean, True if only consider logits
      as the last layers (i.e., exclude ASPP module, decoder module and so on).

  Returns:
    A list of scopes for extra layers.
  """
  if last_layers_contain_logits_only:
    return [_LOGITS_SCOPE_NAME]
  else:
    return [
        _LOGITS_SCOPE_NAME,
        _IMAGE_POOLING_SCOPE,
        _ASPP_SCOPE,
        _CONCAT_PROJECTION_SCOPE,
        _DECODER_SCOPE,
    ]

2. Add a flag in train.py.

flags.DEFINE_boolean('last_layers_contain_logits_only', False,
                     'Only consider logits as last layers or not.')

3. Modify line 295 in train.py as follows.

last_layers = model.get_extra_layer_scopes(
          FLAGS.last_layers_contain_logits_only)